pipeline/transformer/transformer_marc_common/src/test/scala/weco/pipeline/transformer/marc_common/transformers/MarcFormatTest.scala

```diff
@@ -0,0 +1,56 @@
+package weco.pipeline.transformer.marc_common.transformers
+
+import org.scalatest.OptionValues
+import org.scalatest.funspec.AnyFunSpec
+import org.scalatest.matchers.should.Matchers
+import weco.catalogue.internal_model.work.Format.{EBooks, EJournals}
+import weco.pipeline.transformer.marc_common.generators.MarcTestRecord
+import weco.pipeline.transformer.marc_common.models.MarcControlField
+
+class MarcFormatTest extends AnyFunSpec with Matchers with OptionValues {
+  describe(
+    "Extracting format information from MARC Leader/07, 006/06 & 008 material specific details"
+  ) {
+    info("https://www.loc.gov/marc/bibliographic/bdleader.html")
+    info("https://www.loc.gov/marc/bibliographic/bd006.html")
+    info("https://www.loc.gov/marc/bibliographic/bd008s.html")
+
+    it("returns an e-journal format when metadata indicates it") {
```

Comment:
I'd like these test titles to be a bit more descriptive of the metadata that causes the result.  Particularly because the wire feed style of data makes it hard to immediately tell what is pertinent.  Perhaps something like this:

```
describe("when the seventh character of the leader is 's'")
it("returns an e-journal when the sixth character in field 006  'o'"){
}
it("returns nothing for any other value of 006[6]"){
}
}
```
etc.

It would also be possible to drive this from TableDrivenPropertyChecks, which might also help because something like `leader= s"00000ca{$bibLevel} a22000003a 4500"` would show which character(s) are relevant.  However, I'm not sure whether that would be better without actually writing it out and looking at it.

Reply:
I think the table tests are nicer, will do that.